### DISCUSSION TITLE
Examples or tutorials on how to consume headless ui in custom design system

### DISCUSSION THREAD

**Discussion by awkale** [tailwindlabs/headlessui#349] [Show and tell]
I'm really enjoying this component library. Does anyone have good examples on how to best use these components in a design system for reusable components?

I'm able to create components from the docs and style them but I'm not confident I'm doing it correctly for a shared component library. I can't find good examples on the frontend architecture when creating a component library, like should I add `forwardRef`s or how do I expose the current state to the parent component. Any guidance would be great!

**Comment by vvo** [comment]
👍 Showing how to reuse something like the Modal component would be great.

**Comment by its-monotype** [comment]
Any updates?

**Comment by krhubert** [comment]
@RobinMalfait @thecrypticace do you know of any github repo with a design system implemented on top of headless ui you think is good and worth following?

**Reply by mewdev** [reply]
> @RobinMalfait @thecrypticace do you know of any github repo with a design system implemented on top of headless ui you think is good and worth following?

Following!
